syntax = "proto3";

package austinrubiks;

service RubiksMessaging {

    rpc GetColumn(Index) returns (SquareTriple) {}

    rpc GetRow(Index) returns (SquareTriple) {}

    rpc SetColumn(SquareTriple) returns (Ack) {}

    rpc SetRow(SquareTriple) returns (Ack) {}

    rpc Rotate(Direction) returns (Ack) {}

}

message Direction {
    string direction = 1;
}

message Index {
    int32 index = 1;
}

message SquareTriple {
    int32 index = 1;
    string square1 = 2;
    string square2 = 3;
    string square3 = 4;
}

message Ack {
    int32 status = 1;
}